Add contributing templates

Pezmc opened the pull request:
Last week GitHub [announced](
https://github.com/blog/2111-issue-and-pull-request-templates) support for templates for Issues and Pull Requests to projects to try and help contributors add the right details at the start of a thread.

Given the large number of issues/PRs Chosen gets, adding some templates may help make things a little more manageable. 

I've put together a proposed ISSUE_TEMPLATE and PULL_REQUEST_TEMPLATE, which were mostly based on the existing [contributing.md](https://github.com/harvesthq/chosen/blob/master/contributing.md).

Do you think these will be of value? Do you have any proposed changes to my drafts?

kevin-brown commented:
General (quick) recommendations:

- Anything you wouldn't want posted to the ticket (notes to the contributor) should be put in HTML comments, just in case they don't remove them.
- You might want to focus the default issue template on bug reports. While having a template for feature requests is great, they're probably less likely.

koenpunt commented on ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,49 @@
+
+## Bug Report Template
+
+### Subject of the issue
```
Unnecessary, because an issue already has a title/subject field

koenpunt commented on ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,49 @@
+### Subject of the issue
+Describe your issue here.
+
+### Your environment
```
This should probably be "My environment"

Pezmc commented:
I've addressed the comments above, thank you, any further thoughts @pfiller, @kenearley, @stof, @koenpunt, and @tjschuck?

I'll wait until at least two :+1:'s before merging.

koenpunt commented on ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,46 @@
+2. This is the second step
+3. Further steps, etc.
+
+Please provide a working demo, you can use [this template](https://jsfiddle.net/Pezmc/5v3v353z/) as a base.
```
I would prefer an anonymous demo fiddle; https://jsfiddle.net/5v3v353z/

Pezmc commented:
I take my previous comment back, if no one has any further objections/suggestions before EOD I'm going to go ahead and merge. We can always adjust these templates in follow ups.

koenpunt commented:
Me might consider putting these files in a `.github` directory?

Pezmc commented:
@koenpunt Tidied into `.github`

koenpunt commented:
I'm OK with this, but @pfiller will get the final word. But he's not around ATM, so could take a few days.

tjschuck commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,46 @@
+
+Please provide a working demo, you can use [this template](https://jsfiddle.net/hyktf3he) as a base.
+
+### Expected behaviour
```
@Pezmc I think you mean "behavior".

:us: :us: :us: :us: :us: :us: :us: :us: :us:

koenpunt replied:
So patriotic

tjschuck commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,46 @@
+### Expected behaviour
+Tell us what should happen.
+
+### Actual behaviour
```
Ditto.

tjschuck commented:
@pfiller probably doesn't care :)  I'll self-assign.

@Pezmc I agree with including these, but honestly, I think they should be simplified quite a bit.  I'm going to make some additional commits directly to this right now.  If you're :+1: on my changes, then let's just call it good enough for now — you can merge, and we'll :nail_care: from there.

Stay tuned...

koenpunt commented:
We probably want to squash these commits :package:

tjschuck commented:
@koenpunt @Pezmc Okay, I made my edits.  My primary focus was to largely tighten them up — brevity will hopefully encourage adherence (or, y'know, reading them at all).

My most controversial change is probably 19127e3, removing the templating for Feature Requests altogether.  My reasoning was (a.) brevity again, but (b.) these issues aren't generally missing information and needing follow-up, so let's not optimize for them.  But feel free to tell me if you disagree.

MD-rendered versions here:

* [Issue template](https://github.com/harvesthq/chosen/blob/add-contributing-templates/.github/ISSUE_TEMPLATE.md)
* [PR template](https://github.com/harvesthq/chosen/blob/add-contributing-templates/.github/PULL_REQUEST_TEMPLATE.md)

koenpunt commented:
I'd say squash it and ship it :rocket:

tjschuck commented:
I squashed down to two commits: @Pezmc's original authorship and my edits.

Thanks @Pezmc and @koenpunt :heart: :heart: :heart_eyes_cat: :heart: !

pfiller commented:
Hey this is pretty neat. Thanks GitHub!

> @pfiller probably doesn't care :) I'll self-assign.

I _do_ care. I also trust your judgement on what a good template looks like and am happy to see that this moved forward in my absence.

Welcome to the party, @Pezmc :heart:
